.circleci/config.yml

```diff
@@ -20,27 +29,123 @@ jobs:
-    - run: cd ~/trafficserver-7.1.8 && ./configure --prefix=`pwd`/../ts7 && make -j`nproc` && make install
-    - run: echo "banjax/banjax.so $HOME/ts7/libexec/trafficserver/banjax/" >> ~/ts7/etc/trafficserver/plugin.config
-    - run: echo "regex_map http://^127.0.0.1:8080$/ http://127.0.0.1:8000/" >> ~/ts7/etc/trafficserver/remap.config
-    - run: perl -pi -e 's/CONFIG proxy.config.http.cache.required_headers INT .+/CONFIG proxy.config.http.cache.required_headers INT 0/g' ~/ts7/etc/trafficserver/records.config
-    - run: cat ~/ts7/etc/trafficserver/records.config
-    - run: mkdir build7
-    - run: cd build7 && cmake ../ -DCMAKE_CXX_FLAGS="-I$HOME/ts7/include"
-    - run: cd build7 && make -j3
-    - run: mkdir ~/ts7/libexec/trafficserver/banjax
-    - run: cp config/{auth,solver}.html ~/ts7/libexec/trafficserver/banjax/
-    - run: cp build7/banjax.so ~/ts7/libexec/trafficserver/banjax/
-    # Run tests
-    - run: ./build7/unit-tests
-    - run: python ./test/banjax_behavior_test.py --ts-prefix=$HOME/ts7 --ts-layout="standard"
+
+    - run: |
+        apt-get update -qq &&
+        apt-get upgrade -qq
+
+    - run: |
+        apt-get install -qy \
+        apt-transport-https \
+        sudo \
+        curl \
+        rsync \
+        psmisc \
+        python3 \
+        python2.7 \
+        tzdata \
+        unzip \
+        zip \
+        bzip2 \
+        gawk
+
+    - run: |
+        apt-get install -qy \
+        git-core \
+        build-essential \
+        automake \
+        autoconf \
+        libtool \
+        flex \
+        bison \
+        cmake
+
+    - run: |
+        apt-get install -qy \
+        libssl-dev \
+        libhwloc-dev \
+        libcap-dev \
+        tcl-dev \
+        libpcre3-dev \
+        libncurses5-dev \
+        libcurl4-openssl-dev \
+        libyaml-cpp-dev \
+        libyaml-cpp0.5v5 \
+        libre2-dev \
+        libzmq3-dev \
+        libboost-dev \
+        libboost-system-dev \
+        libboost-test-dev \
+        libconfig++-dev
+
+    - run: |
+        groupadd --gid $TSGID $TSGROUP && \
+        useradd --uid $TSUID -s /bin/bash -g $TSGROUP -c "Apache Traffic Server" -M $TSUSER
+
+    - run: |
+        mkdir /build_trafficserver && \
+        cd /build_trafficserver && curl -O $ATS_DLINK
+
+    # XXX ATS_VERSION duplicated here
+    - run: |
+        cd /build_trafficserver && \
+        [[ "$(sha512sum < trafficserver-7.1.8.tar.bz2)" == "$ATS_SHA512  -" ]] && \  # XXX double check this?
+        tar jxf trafficserver-$ATS_VERSION.tar.bz2
+
+    - run: |
+        cd /build_trafficserver/trafficserver-$ATS_VERSION && \
+        ./configure --prefix=$TSPATH \
+        --enable-layout=Apache \
+        --with-user=$TSUSER \
+        --with-group=$TSGROUP \
+        --enable-experimental-plugins \
+        --enable-example-plugins \
+        --with-ncurses \
+        --enable-wccp
+
+    - run: |
+        cd /build_trafficserver/trafficserver-$ATS_VERSION && \
+        make -j$(nproc) && \
```

Comment:
Unfortunately it seems `nproc` returns some crazy big number on cci (like 36 or so). I don't think that was the case before, but using it will only slow down the compilation. In Ouinet we currently use the number 3.